p8e-api/src/test/kotlin/helper/TestUtils.kt

```diff
@@ -45,5 +57,263 @@ class TestUtils {
+                                                    PK.SigningAndEncryptionPublicKeys.newBuilder()
+                                                        .setEncryptionPublicKey(keys.public.toPublicKeyProto())
+                                                        .setSigningPublicKey(keys.public.toPublicKeyProto())
+                                                        .build() // Build Signer
+                                                )
+                                                .setAddress("some-address".toByteString())
+                                                .build() // Build Recital
+                                        )
+                                    )
+                                    .addAllRecords(
+                                        mutableListOf(
+                                            ContractScope.Record.newBuilder()
+                                                .setName("some-name")
+                                                .setHash("AyZYcO1gmfndNHU+v4ltISy+nZb5rdwNjfc5+Q66dbpC3tlfF5Nt79usqSZjIz8h3HoJEYcIz3LE7sM09uTUXg==")
+                                                .setClassname("HelloWorldContract")
+                                                .addAllInputs(
+                                                    mutableListOf(
+                                                        ContractScope.RecordInput.newBuilder()
+                                                            .setName("some-name")
+                                                            .setHash("AyZYcO1gmfndNHU+v4ltISy+nZb5rdwNjfc5+Q66dbpC3tlfF5Nt79usqSZjIz8h3HoJEYcIz3LE7sM09uTUXg==")
+                                                            .setClassname("HelloWorldContract")
+                                                            .setType(ContractScope.RecordInput.Type.PROPOSED)
+                                                            .build() // Build RecordInput
+                                                    )
+                                                )
+                                                .setResult(Contracts.ExecutionResult.Result.PASS)
+                                                .setResultName("some-result-name")
+                                                .setResultHash("some-result-hash")
+                                                .build() // Build Record
+                                        )
+
+                                    )
+                                    .setClassname("HelloWorldContract")
+                                    .build() // Build RecordGroup
+                            )
+                        )
+                        .setLastEvent(lastEvent)
+                        .build() // Build Scope
+                )
+                .setStatus(ContractScope.Envelope.Status.CREATED)
+                .build() // Build Envelope
+        }
     }
-}
+}
```

Comment:
can you configure your editor to add new lines at end of files?

Reply:
done.